### PR TITLE
`Dex` - remove liquidity page

### LINKED ISSUE #145: `Dex` - remove liquidity
#### What would you like to be added:
Remove current liqudity pools
#### Why is this needed:
image

### PR DESCRIPTION
**Pull request by ivan-zynesis**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #145 

#### Additional comments?:
include changes of #205 
image
image

### CODE REVIEW

File: screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      style={[tailwind('m-2 p-3 rounded flex-row justify-center'), buttonColor]}
+      onPress={props.onPress}
+    >
+      <Text style={[tailwind('text-white font-bold')]}>Continue</Text>
+    </TouchableOpacity>
+  )
+}
+
+async function constructSignedRemoveLiqAndSend (whaleAPI: WhaleApiClient, account: WhaleWalletAccount, tokenId: number, amount: BigNumber): Promise<string> {
+  const feeRate = new WhaleFeeRateProvider(whaleAPI)
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx:173]: Fuxing refactored the transaction builder to require fewer params. You can take a look here - https://github.com/DeFiCh/wallet/blob/0dcdca523d14a1e234371dff0e8633f11d74ce06/screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx#L52

File: screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+  return (
+    <View style={tailwind('w-full h-full')}>
+      <ScrollView style={tailwind('w-full flex-col flex-1 bg-gray-100')}>
+        <View style={tailwind('w-full bg-white mt-8')}>
+          <View style={tailwind('w-full flex-row p-4')}>
+            <Text style={tailwind('flex-1')}>Amount of liquidity to remove</Text>
+            <Text style={tailwind('text-right')}>{percentage} %</Text>
+          </View>
+          <AmountSlider
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx:84]: Slider is broken on Android
Screenshot_20210702-101630_Expo Go

File: screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        {props.rhs.map((val, idx) => (<Text key={idx} style={tailwind('font-medium text-right text-gray-500')}>{val}</Text>))}
+      </View>
+    </View>
+  )
+}
+
+function ContinueButton (props: { enabled: boolean, onPress: () => void }): JSX.Element {
+  const buttonColor = props.enabled ? PrimaryColorStyle.bg : { backgroundColor: 'gray' }
+  return (
+    <TouchableOpacity
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx:162]: You can update to PrimaryButton component

File: screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+function CoinAmountRow (props: { symbol: string, amount: BigNumber }): JSX.Element {
+  const TokenIcon = getTokenIcon(props.symbol)
+  return (
+    <View style={tailwind('flex-row items-center border-t border-gray-200 p-2')}>
+      <View style={tailwind('flex-row flex-1 items-center justify-start')}>
+        <TokenIcon style={tailwind('ml-2')} />
+        <Text style={tailwind('ml-2')}>{props.symbol}</Text>
+      </View>
+      <Text style={tailwind('flex-1 text-right text-gray-500 mr-2')}>{props.amount.toString()}</Text>
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx:141]: NumberText

File: screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        maximumValue={100}
+        minimumTrackTintColor='#ff00af'
+        thumbTintColor='#ff00af'
+        onValueChange={(val) => props.onChange(val)}
+      />
+      <TouchableOpacity
+        style={tailwind('ml-4')}
+        onPress={() => props.onChange(100)}
+      >
+        <Text style={tailwind('text-gray-500 text-xs')}>All</Text>
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx:127]: translation

File: screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+function AmountSlider (props: { current: number, onChange: (percentage: number) => void, viewStyle: StyleProp<ViewStyle> }): JSX.Element {
+  return (
+    <View style={[tailwind('flex-row items-center border-t border-gray-200'), props.viewStyle]}>
+      <TouchableOpacity
+        style={tailwind('mr-4')}
+        onPress={() => props.onChange(0)}
+      >
+        <Text style={tailwind('text-gray-500 text-xs')}>None</Text>
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx:113]: translation

File: screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+          <AmountSlider
+            current={Number(percentage)}
+            viewStyle={tailwind('p-4')}
+            onChange={setSliderPercentage}
+          />
+        </View>
+        <View style={tailwind('w-full bg-white mt-8')}>
+          <CoinAmountRow symbol={aSymbol} amount={tokenAAmount} />
+          <CoinAmountRow symbol={bSymbol} amount={tokenBAmount} />
+          <PriceRow lhs='Price' rhs={[`${tokenAPerLmToken.toString()} ${aSymbol}`, `${tokenBPerLmToken.toString()} ${bSymbol}`]} />
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx:93]: translation for Price

File: screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+      console.log(e)
+    })
+  }, [amount])
+
+  return (
+    <View style={tailwind('w-full h-full')}>
+      <ScrollView style={tailwind('w-full flex-col flex-1 bg-gray-100')}>
+        <View style={tailwind('w-full bg-white mt-8')}>
+          <View style={tailwind('w-full flex-row p-4')}>
+            <Text style={tailwind('flex-1')}>Amount of liquidity to remove</Text>
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx:81]: translation

File: screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+
+function ContinueButton (props: { enabled: boolean, onPress: () => void }): JSX.Element {
+  const buttonColor = props.enabled ? PrimaryColorStyle.bg : { backgroundColor: 'gray' }
+  return (
+    <TouchableOpacity
+      disabled={!props.enabled}
+      style={[tailwind('m-2 p-3 rounded flex-row justify-center'), buttonColor]}
+      onPress={props.onPress}
+    >
+      <Text style={[tailwind('text-white font-bold')]}>Continue</Text>
```
Comment (thedoublejay) [inline screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx:167]: translation

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx (hunk 2 above)
Comment (ivan-zynesis) [reply screens/AppNavigator/screens/Dex/DexRemoveLiquidity.tsx:84]: Fixed

**Review by thedoublejay — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**